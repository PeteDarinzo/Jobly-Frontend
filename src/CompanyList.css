

.CompanyList {
  position: relative;
  top: 25px;
  width: 100%;
}

.CompanyList-link {
  text-decoration: none;
  color: black;
}

.CompanyList > div {
  width: 50%;
  margin: 50px auto 0 auto;
}
