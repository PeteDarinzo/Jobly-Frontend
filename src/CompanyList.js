import React, { useState } from "react";
import CompanyCard from "./CompanyCard";
import { Link } from "react-router-dom";
import "./CompanyList.css";
import CompanySearchForm from "./CompanySearchForm";
import { v4 as uuid } from "uuid";
import JoblyApi from "./Api";


const CompanyList = ({ companies }) => {

  const [companyList, setCompanyList] = useState(companies)

  /** Take the filter from the search form, return companies mathing filter query
   *  getAllCompanies takes an object and sets it as data in the request
   *  "name" is the filter to search companies by name
   */
  async function filterCompanies(term) {
    const res = await JoblyApi.getAllCompanies({ name: term });
    setCompanyList(res)
  }

  return (
    <div>
      <h1>All Companies:</h1>
      <CompanySearchForm filter={filterCompanies} />
      {companyList.map(company => (
        <Link to={`companies/${company.handle}`} className="CompanyListLink" key={uuid()}>
          <CompanyCard name={company.name} description={company.description} />
        </Link>
      ))
      }
    </div >
  );
}

export default CompanyList;