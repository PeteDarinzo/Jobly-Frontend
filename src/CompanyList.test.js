import CompanyList from "./CompanyList";
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom"

it("renders home with crashing", () => {
  render(
    <MemoryRouter>
      <CompanyList companies={[]} filter={() => { }} clearFilter={() => { }} />
    </MemoryRouter>
  );
});

it("matches the home snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyList companies={[]} filter={() => { }} clearFilter={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});