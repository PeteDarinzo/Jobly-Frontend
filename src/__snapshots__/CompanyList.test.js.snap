// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the home snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyList"
  >
    <h1>
      All Companies:
    </h1>
    <form
      class="Form shadow"
    >
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="search"
        >
          Search Companies
        </label>
        <input
          class="form-control"
          id="search"
          name="search"
          placeholder="Enter search term"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <button
          class="m-1 btn btn-primary"
          type="button"
        >
          Filter Companies
        </button>
        <button
          class="m-1 btn btn-secondary"
          type="button"
        >
          Clear Filters
        </button>
      </div>
    </form>
    <div
      class="CompanyList-list"
    />
  </div>
</DocumentFragment>
`;
